### ISSUE TITLE
Writing into a moved-from value is fine, but not reading what was just written?

### ISSUE THREAD

**Issue by scottmcm** [rust-lang/rust#40468]
I noticed this weird situation (1.15.1 and nightly):
```
#[derive(Debug, Clone)]
struct Point { x: String, y: String }

fn main() {
    let mut p = Point { x: "1".into(), y: "2".into() };
    let _x = p;
    p.x = "4".into(); // this line is fine
    let _z = p.x; // this is a use after move
}
```

It feels like either:

- The assignment into the moved value should cause a diagnostic, or
- The assignment to the field should make its use legal again

The latter is interesting in that it makes moving a struct behave like moving all its fields, but is also a bit odd.

(Note that adding an `impl Drop for Point` to the above _does_ already give the expected ```error[E0383]: partial reinitialization of uninitialized structure `p`.```, but that doesn't conflict with the "move all the fields" interpretation, since you can't move fields from something that `needs_drop` anyway.)

If nothing else, the example in https://doc.rust-lang.org/error-index.html#E0383 needs updating, as it doesn't produce the error it claims to.

**Comment by oli-obk** [comment]
duplicate of #21232
